### PR TITLE
Dev/ssl cipher support

### PR DESCRIPTION
**Pull request by eallison91**
Allows optional specifying of which ssl ciphers the server should support. Default behavior is unchanged.

Created from https://github.com/puma/puma/pull/1476 which was inadvertently closed.

### CODE REVIEW

File: test/test_binder.rb [hunk 1]
```diff
@@ -28,4 +28,32 @@ def test_localhost_addresses_dont_alter_listeners_for_ssl_addresses
...
+
+    ssl = @binder.instance_variable_get(:@ios)[0]
+    ctx = ssl.instance_variable_get(:@ctx)
+    assert_equal(ssl_cipher_filter, ctx.ssl_cipher_filter)
+  end
+
+  def test_binder_parses_jruby_ssl_options
+    skip unless Puma.jruby?
+
+    keystore = File.expand_path "../../examples/puma/keystore.jks"
```
Comment (mirceal) [inline test/test_binder.rb]: missing ", __FILE__" from end of line

**Comment by eallison91** [comment]
https://github.com/puma/puma/issues/996

**Comment by eallison91** [comment]
Any update here? Many of the failures do not appear to be related to my changes and I'd like to resolve this.

If there's a way to run the travis and/or appveyor tests locally, I would welcome any information on it. At this point, all tests pass locally.

**Comment by nateberkopec** [comment]
Thanks for the PR. I'm taking some time off during the holidays and probably won't be able to merge until 2018.

**Comment by mirceal** [comment]
any update on this?

**Comment by arzavj** [comment]
Would love to see this in soon too!

**Comment by evanphx** [comment]
The Travis errors appear to be from the compiler being updated, we'll need to fix those but this PR didn't cause that.

**Pull request merged**